### PR TITLE
libxml2 and libxslt: build against python3 by default

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
cc @NixOS/darwin-maintainers. Should we use Python 3 for the bootstrapping Python or is that orthogonal to this issue?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -12524,7 +12524,9 @@ in
 
   libxmi = callPackage ../development/libraries/libxmi { };
 
-  libxml2 = callPackage ../development/libraries/libxml2 { };
+  libxml2 = callPackage ../development/libraries/libxml2 {
+    python3 = if stdenv.isDarwin then python2 else python3;
```
Comment (edolstra) [inline pkgs/top-level/all-packages.nix]: Passing `python2` via a function argument named `python3` is super confusing. Better to just call the function argument `python`. Even better is to make it work the same on macOS...

**Comment by chkno** [comment]
After this change, I get this error: 
```
$ nix-shell -p shared-mime-info -I nixpkgs=.
...
Traceback (most recent call last):
  File "/nix/store/b2cms2q9bmrhp1as93m8vbkjq9r6zn5y-itstool-2.0.2/bin/itstool", line 1545, in <module>
    out = file(opts.output, 'w')
NameError: name 'file' is not defined
make[2]: *** [Makefile:1179: freedesktop.org.xml] Error 1
make[2]: Leaving directory '/build/shared-mime-info-1.13.1'
make[1]: *** [Makefile:731: all-recursive] Error 1
make[1]: Leaving directory '/build/shared-mime-info-1.13.1'
make: *** [Makefile:455: all] Error 2
builder for '/nix/store/40rwsy7q9hbiqhgi5ck5kibmirmywv8k-shared-mime-info-1.13.1.drv' failed with exit code 2
error: build of '/nix/store/40rwsy7q9hbiqhgi5ck5kibmirmywv8k-shared-mime-info-1.13.1.drv' failed
```

I notice this error when I try to run any NixOS test, such as `nix-build nixos/tests/login.nix`.

Reverting 8d926467be6438fd88b19b38389ad1fdfc472f4d or makes the error go away, as does dropping itstool back to python2 as in #72335.

[itstool says it does not support Python 3 until version 2.0.3](http://itstool.org/download.html), but bumping its version isn't straightforward because [itstool/default.nix says 2.0.3+ breaks the build of gnome3.gnome-desktop](https://github.com/NixOS/nixpkgs/blob/8d926467be6438fd88b19b38389ad1fdfc472f4d/pkgs/development/tools/misc/itstool/default.nix).

**Comment by worldofpeace** [comment]
@chkno itstool isn't even used in gnome-desktop anymore, it's likely this issue has been resolved already and we can update itstool, or maybe there's a patch needed in libxml2.

**Comment by worldofpeace** [comment]
Yeah, distro's have this patch from fedora 
* https://git.archlinux.org/svntogit/packages.git/tree/trunk/itstool-2.0.5-fix-crash-wrong-encoding.patch?h=packages/itstool

to fix the crash. Maybe it has been committed in a newer version.

**Comment by jtojnar** [comment]
@worldofpeace unfortunately gnome-desktop was not the only affected package, the cause is not fixed https://bugzilla.gnome.org/show_bug.cgi?id=789714

**Comment by worldofpeace** [comment]
@jtojnar https://src.fedoraproject.org/rpms/libxml2/blob/master/f/libxml2-2.9.8-python3-unicode-errors.patch?

**Comment by jtojnar** [comment]
Yes, that seems to have come from https://bugzilla.opensuse.org/show_bug.cgi?id=1065270. It is definitely better but as https://bugzilla.gnome.org/show_bug.cgi?id=789714#c4 says, it is just a hack for broken libxml2. I wonder why is this not considered a critical security vulnerability.

**Comment by worldofpeace** [comment]
> Yes, that seems to have come from https://bugzilla.opensuse.org/show_bug.cgi?id=1065270. It is definitely better but as https://bugzilla.gnome.org/show_bug.cgi?id=789714#c4 says, it is just a hack for broken libxml2. I wonder why is this not considered a critical security vulnerability.

It seems upstream doesn't find it concerning
* https://gitlab.gnome.org/GNOME/libxml2/issues/12

**Pull request merged**